StreamChatClient_v3/WebSocketClient/WebSocketPingController.swift

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+class WebSocketPingController {
+    typealias PingAction = () -> Void
+    typealias ForceReconnectAction = () -> Void
+    
+    /// The time interval to ping connection to keep it alive.
+    static let pingTimeInterval: TimeInterval = 25
+    static let pongTimeoutTimeInterval: TimeInterval = 3
+    
+    private let timerType: Timer.Type
+    private let timerQueue: DispatchQueue
+    
+    /// The timer used for scheduling `ping` calls
+    private lazy var pingTimerControl: RepeatingTimerControl =
+        timerType.scheduleRepeating(timeInterval: Self.pingTimeInterval, queue: timerQueue) { [weak self] in self?.sendPing() }
+    
+    /// The pong timeout timer.
+    private var pongTimeoutTimer: TimerControl?
+    private let ping: () -> Void
+    private let forceReconnect: () -> Void
+    
+    /// Creates a ping controller.
+    /// - Parameters:
+    ///   - timerType: a timer type.
+    ///   - timerQueue: a timer dispatch queue.
+    ///   - ping: an action for `WebSocketClient` to send a ping.
+    ///   - forceReconnect: an action for `WebSocketClient` to force disconnect and reconnect.
+    init(timerType: Timer.Type,
+         timerQueue: DispatchQueue,
+         ping: @escaping PingAction,
+         forceReconnect: @escaping ForceReconnectAction) {
+        self.timerType = timerType
+        self.timerQueue = timerQueue
+        self.ping = ping
+        self.forceReconnect = forceReconnect
+    }
+    
+    func connectionStateDidChange(_ connectionState: ConnectionState) {
+        cancelPongTimeoutTimer()
+        
+        if connectionState.isConnected {
+            log.info("🏓 Resume ping timer")
```

Comment:
Please don't use emojis in log. Our logger supports emojis in the formatter.

Reply:
How I can setup emoji in our Logger for ping/pong only?

Reply:
@buh 
```swift
public class PingPongEmojiFormatter: LogFormatter {
    public func format(logDetails: LogDetails, message: String) -> String {
        if logDetails.message.contains("WebSocket Ping") {
            return "🏓➡️" + " " + message
        } else if logDetails.message.contains("WebSocket Pong") {
            return "⬅️🏓" + " " + message
        } else {
            return message
        }
    }
}
```
add this as a new file to `Logger.Formatters` and use it in sample project